code/game/objects/items/weapons/grenades.dm

```diff
@@ -292,64 +281,145 @@
 		new /obj/item/device/multitool(src) // used to adjust the chemgrenade's signaller
 		new /obj/item/device/assembly/signaler(src)
 		return
-// --------------------------------------
-// Clusterbuster Variable Payload Grenade
-//   Adapted from flashbang/clusterbang
-// --------------------------------------
-
+////////////////////
+//Clusterbang
+////////////////////
 /obj/item/weapon/grenade/clusterbuster
-	desc = "This highly intimidating bunch of hardware seems eager to be let loose."
-	name = "Clusterbang"
+	desc = "Use of this weapon may constiute a war crime in your area, consult your local captain."
+	name = "clusterbang"
 	icon = 'icons/obj/grenade.dmi'
 	icon_state = "clusterbang"
-	var/payload = /obj/item/weapon/grenade/flashbang
+	var/payload = /obj/item/weapon/grenade/flashbang/cluster
 
+/obj/item/weapon/grenade/clusterbuster/prime()
+	update_mob()
+	var/numspawned = rand(4,8)
+	var/again = 0
 
-// Subtypes
+	for(var/more = numspawned,more > 0,more--)
+		if(prob(35))
+			again++
+			numspawned--
+
+	for(var/loop = again ,loop > 0, loop--)
+		new /obj/item/weapon/grenade/clusterbuster/segment(loc, payload)//Creates 'segments' that launches a few more payloads
+
+	new /obj/effect/payload_spawner(loc, payload, numspawned)//Launches payload
+
+	playsound(loc, 'sound/weapons/armbomb.ogg', 75, 1, -3)
+
+	qdel(src)
+
+
+//////////////////////
+//Clusterbang segment
+//////////////////////
+/obj/item/weapon/grenade/clusterbuster/segment
+	desc = "A smaller segment of a clusterbang. Better run."
+	name = "clusterbang segment"
+	icon = 'icons/obj/grenade.dmi'
+	icon_state = "clusterbang_segment"
+
+/obj/item/weapon/grenade/clusterbuster/segment/New(var/loc, var/payload_type = /obj/item/weapon/grenade/flashbang/cluster)
+	..()
+	icon_state = "clusterbang_segment_active"
+	payload = payload_type
+	active = 1
+	walk_away(src,loc,rand(1,4))
+	spawn(rand(15,60))
+		prime()
+
+
+/obj/item/weapon/grenade/clusterbuster/segment/prime()
+
+	new /obj/effect/payload_spawner(loc, payload, rand(4,8))
+
+	playsound(loc, 'sound/weapons/armbomb.ogg', 75, 1, -3)
+
+	qdel(src)
+
+//////////////////////////////////
+//The payload spawner effect
+/////////////////////////////////
+/obj/effect/payload_spawner/New(var/turf/newloc,var/type, var/numspawned as num)
+
+	for(var/loop = numspawned ,loop > 0, loop--)
+		var/obj/item/weapon/grenade/P = new type(loc)
```

Comment:
This only working for grenades seems to be entirely arbitrary, all it has to do is run an istype() for grenades after the walk_away proc.